pallets/gear/src/builtin.rs

```diff
@@ -0,0 +1,70 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+use gear_core::ids::{BuiltinId, ProgramId};
+
+/// A trait representing a registry that provides methods to lookup a builtin actor
+pub trait BuiltinRouter {
+    type QueuedDispatch;
+    type Output;
```

Comment:
I guess at least Output may be removed in favour of just JournalNote.
And anyway if you disagree it more likely that dispatch will -> Option<Self::Output> where Output = Vec<_>

Reply:
Probably this makes sense: abstracting away the `JournalNote` type is unnecessary here.